materialize-boilerplate/apply.go

```diff
@@ -178,10 +178,12 @@ func ApplyChanges(ctx context.Context, req *pm.Request_Apply, applier Applier, i
 					}
 
 					newRequired := projection.Inference.Exists == pf.Inference_MUST && !slices.Contains(projection.Inference.Types, pf.JsonTypeNull)
-					if !existingField.Nullable && !newRequired && !existingField.HasDefault {
-						// The existing field is not nullable and does not have a default value, but
-						// the proposed projection for the field is nullable. The existing field
-						// will need to be modified to be made nullable.
+					newlyNullable := !existingField.Nullable && !newRequired
+					projectionHasDefault := projection.Inference.DefaultJson != nil
+					if newlyNullable && !existingField.HasDefault && !projectionHasDefault {
```

Comment:
why do we check for whether the existing field has a default value? If someone updates a collection and projection to make a field nullable, but at the same time add a projection with a default value, we should still allow them to keep their column as non-nullable, no?

Reply:
`existingField` refers to the actual column in the destination system. So if they have altered or pre-created their database table with a column that is not null and has a default value, we allow them to keep that column as non-nullable even if the Flow collection field is nullable and does not have a default value.

The situation you're describing is allowed - if someone updates a collection and makes a field nullable but adds a default value to it, we won't drop nullability on the column. We only will drop nullability on a column if the column itself doesn't have a default value, and our Flow projection doesn't have a default value.